backend/src/main/java/com/bakdata/conquery/sql/conversion/model/StepJoiner.java

```diff
@@ -71,6 +69,41 @@ private static TableLike<Record> constructJoinedTable(List<QueryStep> queriesToJ
 		return joinedQuery;
 	}
 
+	private static Field<Object> coalescePrimaryColumns(List<QueryStep> querySteps) {
+		List<Field<?>> primaryColumns = querySteps.stream()
+												  .map(queryStep -> queryStep.getQualifiedSelects().getPrimaryColumn())
+												  .collect(Collectors.toList());
+		return DSL.coalesce(primaryColumns.get(0), primaryColumns.subList(1, primaryColumns.size()).toArray())
+				  .as(PRIMARY_COLUMN_NAME);
+	}
+
+	private static Optional<ColumnDateRange> extractValidityDates(List<QueryStep> querySteps) {
+		// TODO: date aggregation...
```

Comment:
noch offen?

Reply:
Ja, dieser PR umfasst nur die Aggregation auf Concept-Level - die Aggregation von mehreren Konzepten ist in einem extra PR, der auf diesem hier basiert. Das wäre sonst zu groß geworden.